components/event-bus/cmd/event-bus-publish-knative/handlers/publish.go

```diff
@@ -239,3 +278,75 @@ func sendJSONError(w http.ResponseWriter, err *api.Error) error {
 	w.WriteHeader((*err).Status)
 	return json.NewEncoder(w).Encode(*err)
 }
+
+func handleKnativePublishRequestV2(w http.ResponseWriter, r *http.Request, knativeLib *knative.KnativeLib,
+	knativePublisher *publisher.KnativePublisher, context *api.TraceContext, opts *opts.Options) (*Message,
+	*string, *string, *api.Error, string) {
+	// validate the http request
+	event, err := validators.ValidateRequestV2(r)
+	if err != nil {
+		log.Printf("validate request failed: %v", err)
+		_ = sendJSONError(w, err)
+		return nil, nil, nil, err, publisher.FAILED
+	}
+
+	//set source-id from the headers if missing in the payload
```

Comment:
Please remove these lines if not needed.

Reply:
I'll